### PR TITLE
Added fuzzing

### PR DESCRIPTION
**Pull request by StephanKa**
- use example from libFuzzer itself
- added configuration in CMakePreset
- added example GitHub pipeline

### CODE REVIEW

File: fuzz_test/fuzz_tester.cpp [hunk 1]
```diff
@@ -1,20 +1,18 @@
-#include <fmt/format.h>
-#include <iterator>
-#include <utility>
+#include <stdint.h>
```
Comment (arnemertz) [inline fuzz_test/fuzz_tester.cpp]: cstdint, cstddef

File: fuzz_test/fuzz_tester.cpp [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-    for (std::size_t offset = 0; offset < Size; ++offset) { value += static_cast<int>(*std::next(Data, static_cast<long>(offset))) * scale; }
-    return value;
+// example comes from here: https://github.com/google/fuzzing/blob/master/tutorial/libFuzzer/fuzz_me.cc
+// Documentation can be found here: https://llvm.org/docs/LibFuzzer.html
+bool FuzzMe(const uint8_t *Data, size_t DataSize) {
+    return DataSize >= 3 &&
+           Data[0] == 'F' &&
+           Data[1] == 'U' &&
+           Data[2] == 'Z' &&
+           Data[3] == 'Z';  // :‑<
```
Comment (arnemertz) [inline fuzz_test/fuzz_tester.cpp:14]: access violation when size == 3?

Comment (arnemertz) [reply fuzz_test/fuzz_tester.cpp:14]: if that's on purpose, there should be very clear documentation why.

File: fuzz_test/fuzz_tester.cpp (hunk 1 above)
Comment (StephanKa) [reply fuzz_test/fuzz_tester.cpp]: I don't wanted to change the original test from libFuzzer itself.

File: fuzz_test/fuzz_tester.cpp (hunk 2 above)
Comment (StephanKa) [reply fuzz_test/fuzz_tester.cpp:14]: added comment that this is on purpose

**Review by arnemertz — APPROVED** [review]

**Pull request merged**